File: src/main/java/model/MarkdownText.java
package model;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

import java.io.Serializable;

@DatabaseTable(tableName = "markdown_texts")
public class MarkdownText extends Model implements Serializable {
    public static final String NAME_FIELD_NAME = "name";
    public static final String ID_FIELD_NAME = "id";

    @SuppressWarnings("unused")
    @DatabaseField(columnName = NAME_FIELD_NAME)
    private String name;

    @SuppressWarnings("unused")
    @DatabaseField
    private String data;

}

File: src/main/kotlin/model/MarkdownText.kt
package model

import org.jooq.DSLContext
import org.jooq.generated.tables.MarkdownTexts.MARKDOWN_TEXTS
import service.DbService

import java.io.Serializable
import javax.persistence.Column

class MarkdownText : Model(), Serializable {
    @Column(name="NAME") var name: String? = null
    @Column(name="DATA") var data: String? = null

    override fun save() {
        DbService.execute { context: DSLContext ->
            if (id == null) {
                context.insertInto(MARKDOWN_TEXTS, MARKDOWN_TEXTS.NAME, MARKDOWN_TEXTS.DATA)
                        .values(name, data)
                        .execute()
            } else {
                context.update(MARKDOWN_TEXTS)
                        .set(MARKDOWN_TEXTS.NAME, name)
                        .set(MARKDOWN_TEXTS.DATA, data)
                        .where(MARKDOWN_TEXTS.ID.eq(id))
                        .execute()
            }
        }
    }

    fun sanitizedCopy(): MarkdownText {
        val obj: MarkdownText = MarkdownText()
        obj.name = this.name
        obj.data = this.data
        return obj
    }

    companion object {
        fun loadByName(name: String): MarkdownText? {
            return DbService.query { context: DSLContext ->
                context.select()
                       .from(MARKDOWN_TEXTS)
                       .where(MARKDOWN_TEXTS.NAME.eq(name))
                       .fetchAnyInto(MarkdownText::class.java)
            }
        }
    }
}
